src/views/instances/details/view-instance-menu-item.tsx

```diff
@@ -0,0 +1,28 @@
+import { MenuItem, IconLaunch16 } from '@dhis2/ui'
+import type { FC } from 'react'
+
+type ViewInstanceMenuItemProps = {
+    groupName: string
+    name: string
+    stackName: string
+}
+
+export const ViewInstanceMenuItem: FC<ViewInstanceMenuItemProps> = ({ groupName, name, stackName }) => {
+    const getBaseURL = () => {
+        switch (stackName) {
+            case 'pgadmin':
+                return `${groupName}.im.dhis2.org/${name}-pgadmin`
+            case 'dhis2-core':
+                return `${groupName}.im.dhis2.org/${name}`
+            default:
+                return `${groupName}.im.dhis2.org/${name}`
+        }
+    }
+
+    const handleClick = () => {
+        const url = `https://${getBaseURL()}`
+        window.open(url, '_blank', 'noopener,noreferrer')
+    }
+
+    return <MenuItem dense label="Open Instance" icon={<IconLaunch16 />} onClick={handleClick} />
+}
```

Comment:
No need for a function (unless this causes performance issues, which it most likely never will).
Makes it easier to read:

```suggestion
export const ViewInstanceMenuItem: FC<ViewInstanceMenuItemProps> = ({
    groupName,
    name,
    stackName,
}) => {
    const host = `${groupName}.im.dhis2.org`
    const path = stackName ? `${name}-pgadmin` : name
    const url = `${host}/${path}`

    return (
        <MenuItem
            dense
            label="Open Instance"
            icon={<IconLaunch16 />}
            onClick={() => window.open(url, '_blank', 'noopener,noreferrer')}
        />
}
```